pallets/asset-manager/src/lib.rs

```diff
@@ -320,7 +313,9 @@ pub mod pallet {
 				AssetIdLocation::<T>::contains_key(&asset_id),
 				Error::<T>::UpdateNonExistAsset
 			);
-			AssetIdMetadata::<T>::insert(&asset_id, &metadata);
+
+			T::AssetRegistrar::update_asset_metadata(asset_id, metadata.clone().into())?;
```

Comment:
Despite we use `transactional`, it is still a good practice to put the code that could fail before the code should never fail.